projects/packages/publicize/src/social-image-generator/class-setup.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Setup class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Social_Image_Generator;
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * Class for setting up Social Image Generator-related functionality.
+ */
+class Setup {
+	/**
+	 * Initialise SIG-related functionality.
+	 */
+	public function init() {
+		add_action( 'save_post', array( $this, 'generate_token' ) );
+	}
+
+	/**
+	 * Set an image token.
+	 *
+	 * @param int    $post_id The post ID to set the token for.
+	 * @param string $token The image token for the post.
+	 */
+	public function set_token( $post_id, $token ) {
+		$social_options = get_post_meta( $post_id, \Automattic\Jetpack\Publicize\Publicize::POST_JETPACK_SOCIAL_OPTIONS, true );
+
+		if ( empty( $social_options ) ) {
+			$social_options = array();
+		}
+
+		$updated_options = array_merge_recursive( $social_options, array( 'image_generator_settings' => array( 'token' => sanitize_text_field( $token ) ) ) );
```

Comment:
Each time I save my post this adds another token to an array. The first time `token` has a string value but then it becomes an array on subsequent calls.

Reply:
Ah good catch! I used `array_merge_recursive` instead of `array_replace_recursive` which caused this. Fixed in https://github.com/Automattic/jetpack/pull/29118/commits/ab8e1a2fae5c739e25aaa206533e0f21cea5b54a. I've also added an extra test to prevent this from happening again.

Reply:
Great. Good call on the regression test too